nuxeo-services/nuxeo-platform-usermanager-api/src/main/java/org/nuxeo/ecm/platform/usermanager/UserManager.java

```diff
@@ -52,12 +52,24 @@
 
     boolean validatePassword(String password);
 
+    /**
+     * Calls {@link #getPrincipal(String, boolean)} to retrieve a principal and its references, given its username
+     */
+    default NuxeoPrincipal getPrincipal(String username) {
+        return getPrincipal(username, true);
+    }
+
     /**
      * Retrieves the principal with the given username or null if it does not exist.
      * <p>
      * Can build principals for anonymous and virtual users as well as for users defined in the users directory.
+     *
+     * @param username is the name of the entry in the user directory
+     * @param fetchReferences controls if the references (groups) of the user will be fetched
+     *
+     * @since 11.1
      */
-    NuxeoPrincipal getPrincipal(String username);
+    NuxeoPrincipal getPrincipal(String username, boolean fetchReferences);
```

Comment:
This is a new required API that a customer implementing their own `UserManager` wouldn't have, therefore their code will be broken.

When backward-compat is required, the `default` methods must be present on the new APIs, not on those whose semantics gets extended...

Reply:
Yes, big mistake. I'll fix that.